lib/vagrant-openshift/action/run_customer_diagnostics_tests.rb

```diff
@@ -0,0 +1,79 @@
+#--
+# Copyright 2016 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#++
+
+module Vagrant
+  module Openshift
+    module Action
+      class RunCustomerDiagnosticsTests
+        include CommandHelper
+
+        @@SSH_TIMEOUT = 4800
+
+        def initialize(app, env, options)
+          @app = app
+          @env = env
+          @options = options.clone
+        end
+
+        def run_tests(env, cmds, as_root=true)
+          tests = ''
+          cmds.each do |cmd|
+            tests += "
+echo '***************************************************'
+echo 'Running #{cmd}...'
+time #{cmd}
+echo 'Finished #{cmd}'
+echo '***************************************************'
+"
+          end
+          cmd = %{
+set -e
+pushd #{Constants.build_dir}/openshift-ansible/ >/dev/null
+export PATH=$GOPATH/bin:$PATH
+#{tests}
+popd >/dev/null
+        }
+          exit_code = 0
+          if as_root
+            _,_,exit_code = sudo(env[:machine], cmd, {:timeout => 60*60*4, :fail_on_error => false, :verbose => false})
+          else
+            _,_,exit_code = do_execute(env[:machine], cmd, {:timeout => 60*60*4, :fail_on_error => false, :verbose => false})
+          end
+          exit_code
+        end
+
+        #
+        # All env vars will be added to the beginning of the command like VAR=1 make test
+        #
+        def call(env)
+          @options.delete :logs
+
+          cmd_env = []
+
+          if @options[:envs]
+            cmd_env += @options[:envs]
+          end
+
+          cmd_env << 'go test ./test/integration/... -v'
```

Comment:
For ease of development I would suggest a different API: from the `vagrant-openshfit` command line, define which `make` target in the repo to execute as an argument to this command. You could default it to `make test` for ease of use. This way not only do you have a simple and stable API with the repo (unless you change `make test` to do something other than test, of course) but also as your tests grow and change you have to make no modification to this repository. This is a huge pain point for people in Origin -- when you have a PR that needs to land at the same time as test code updates to Vagrant-OpenShift it is tricky and sometimes even a circular dependency. Furthermore, you'll integrate with my tool in the future for free if you choose `make` as your API today, since I've chosen that as the API between my tool and repositories.